dhis-2/dhis-web/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/SystemSettingController.java

```diff
@@ -184,39 +250,42 @@ public void getSystemSettingsJsonP( @RequestParam( value = "key", required = fal
         @RequestParam( defaultValue = "callback" ) String callback, HttpServletResponse response )
         throws IOException
     {
-        Set<SettingKey> settingKeys = null;
-
-        if ( keys != null )
-        {
-            keys.removeIf( systemSettingManager::isConfidential );
-            settingKeys = keys.stream().map( key -> SettingKey.getByName( key ) ).filter( settingKeyOpt -> settingKeyOpt.isPresent() )
-                .map( settingKeyOpt -> settingKeyOpt.get() ).collect( Collectors.toSet() );
-        }
+        Set<SettingKey> settingKeys = getSettingKeysToFetch( keys );
 
         response.setContentType( MediaType.APPLICATION_JSON_VALUE );
         response.setHeader( ContextUtils.HEADER_CACHE_CONTROL, CacheControl.noCache().cachePrivate().getHeaderValue() );
         renderService.toJsonP( response.getOutputStream(), getSystemSettings( settingKeys ), callback );
     }
 
-    private Map<String, Serializable> getSystemSettings( Set<SettingKey> keys )
+    private Set<SettingKey> getSettingKeysToFetch( Set<String> keys )
     {
-        Map<String, Serializable> value;
-        if ( keys != null && !keys.isEmpty() )
+        Set<SettingKey> settingKeys;
+
+        if ( keys != null )
         {
-            value = systemSettingManager.getSystemSettings( keys );
+            keys.removeIf( systemSettingManager::isConfidential );
+            settingKeys = keys.stream().map( key -> SettingKey.getByName( key ) ).filter( settingKeyOpt -> settingKeyOpt.isPresent() )
+                .map( settingKeyOpt -> settingKeyOpt.get() ).collect( Collectors.toSet() );
```

Comment:
if you want, you can shorten it even more:

```
settingKeys = keys.stream().map( SettingKey::getByName ).filter( Optional::isPresent ).map( Optional::get )
                .collect( Collectors.toSet() );
```

Reply:
I somehow missed complaint from Intellij and only copy-pasted the code. Fixed now.